lib/publish.js

```diff
@@ -48,7 +48,14 @@ module.exports = async (pluginConfig, context) => {
         const label = asset.label ? template(asset.label)(context) : undefined;
         const _url = asset.url;
         if (_url) {
-          assetsList.push({label, rawUrl: _url, type, filepath});
+          let _label = label;
+          if (!label) {
+            const parsedUrl = new URL(_url);
+            const file = pathlib.basename(parsedUrl.pathname);
+            _label = file ? file : 'Unknown asset';
```

Comment:
I don't like the fallback here. What about falling back to parsedUrl.hostname?

Also can any of these throw an exception that we should handle?

Reply:
You mean replacing 'Unknown asset' by `parsedUrl.hostname`?  
Actually I am thinking any better handling here too since the link name must be unique within the release.  

So when user have urls like below will cause exception during publish
https://server1.com/1.txt
https://server2.com/1.txt

Reply:
We could include the index from `.map`, e.g.:
```js
_label = file ? file : `${parsedUrl.hostname}_${index}`;
```